File: 0057-insert-interval/0057-insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        ans = []
        i=0
        while i<len(intervals) and intervals[i][1]<newInterval[0]:
            ans.append(intervals[i])
            i+=1
        
        while i<len(intervals) and newInterval[1]>=intervals[i][0]:
            newInterval[0] = min(newInterval[0],intervals[i][0])
            newInterval[1] = max(newInterval[1],intervals[i][1])
            i+=1
        
        ans.append(newInterval)

        while i<len(intervals):
            ans.append(intervals[i])
            i+=1
        return ans
        

File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        
        // first I have to check at what interval I need to insert this element 
        vector<vector<int>> ans;
        int i=0;
        while(i<intervals.size() && intervals[i][1]<newInterval[0]){
            ans.push_back(intervals[i]);
            i++;
        }

        // Now we will inert the interval
        while(i<intervals.size() && intervals[i][0]<=newInterval[1]){
            newInterval[0] = min(intervals[i][0],newInterval[0]);
            newInterval[1] = max(intervals[i][1],newInterval[1]);
            i++;
        }

        ans.push_back(newInterval);

        while(i<intervals.size()){
            ans.push_back(intervals[i]);
            i++;
        }

        return ans;
    }
};